### ISSUE TITLE
Login via proxy

### ISSUE THREAD

**Issue by Metz01** [meeting-room-booking-system/mrbs-code#3754]
Hi, I was wandering if there is a way to login to my mrbs service (that is login protected) using a proxy of some sort,. I need to se the content of the page in a  digital signage service, where I can't login directly on the displayed page, how can I work around this? 
And one other strange question, using the config file is it possible to make the site reachable only by the lan of the server containing the mrbs system?

**Comment by campbell-m** [comment]
> Hi, I was wandering if there is a way to login to my mrbs service (that is login protected) using a proxy of some sort,. I need to se the content of the page in a digital signage service, where I can't login directly on the displayed page, how can I work around this?

How are you launching the digital sign in the first place?  How do you enter the URL?

> And one other strange question, using the config file is it possible to make the site reachable only by the lan of the server containing the mrbs system?

Not by using the config file.  You may be able to modify the MRBS code to check for the remote IP address, but it's probably better to use .htaccess.

**Comment by Metz01** [comment]
> How are you launching the digital sign in the first place? How do you enter the URL?
The digital signage has a embedded html which I tried to use inserting some js to fill the login form and send the request but it didn't work. The next thing was to use a proxy with flask but still didn't work.

Now I'm trying to use the multisite feature to have an access to the site without a login, but I'm stuck since it redirects me to the login page. In the main config file there is:
`$auth["type"] = "ldap";
$auth['session'] = 'php';
$auth['deny_public_access'] = TRUE;
`
and in the new config:

`$auth["type"] = "none";
$auth['session'] = 'php';
$auth['deny_public_access'] = FALSE;`

but whatever I try it takes me to the login page (where you can login with whatever random credential). Any idea on how to fix this?

**Comment by campbell-m** [comment]
If you have `$auth['deny_public_access'] = FALSE;`  - no matter what the authentication type - you should be able to see the index.php page without having to log in.

Are you sure that you have got multisite set up properly?  For your digital signage site you need to define a sitename, eg 'sign' and then create a file sites/sign/config.inc.php that just contains

~~~php
<?php
$auth['deny_public_access'] = FALSE;
~~~

In your main config file you need to set `$multisite = true;`.

Then to access your digital signage site point your browser at `index.php?site=sign`.

**Comment by Metz01** [comment]
Tried that but still it redirects me to the login

**Comment by campbell-m** [comment]
And has the login got the `site=sign` in the URL query string?  Or maybe you can't see the URL, in which case try testing it in another browser before moving to digital signage.

**Comment by campbell-m** [comment]
Do you mean the SAML login page or the LDAP login page?

**Comment by Metz01** [comment]
Now Is redirecting to the saml page

**Comment by campbell-m** [comment]
How is it going to the SAML page?  Have you got `$auth['type'] = 'saml';` anywhere in your config files?

**Comment by Metz01** [comment]
It's only in the config file used for the saml authentication site

**Comment by campbell-m** [comment]
What SAML authentication site?  I thought you just had 

`$auth["type"] = "ldap"; $auth['session'] = 'php'; $auth['deny_public_access'] = TRUE; ` and then in your open site

`$auth["type"] = "none"; $auth['session'] = 'php'; $auth['deny_public_access'] = FALSE;`?

**Comment by Metz01** [comment]
Sorry, I've made some mess here. Let me sum up:
I've 2 sites already working correctly:
The First One gas a config file with the saml authentication which Is called by default or by using site=external
The second One gas a config file for the LDAP login which Is reachable by site=internal.

Now I'm trying to add a third One (site=open) which has no login Page. In this config file I tried to put the auth type to none, auth session to PHP and deny_public_access to false, but It would redirect me to a "fake"  login, where whatever credential I put It granted me access. Istead, if I Just write a config file with deny_public_access to false It redirects me to the saml authentication Page that I use in the First site (site=external).

I Hope it's more clear now, Sorry for the mess

**Comment by campbell-m** [comment]
Which version of MRBS are you using?

**Comment by Metz01** [comment]
1.11.4

**Comment by campbell-m** [comment]
And are you pointing your browser at index.php, rather than any other page?

**Comment by Metz01** [comment]
Yes, even explicity declaring /index.php?site=open It redirects me

**Comment by campbell-m** [comment]
Can you post your copy of mrbs_auth.inc here please?

**Comment by Metz01** [comment]
Right guess! To ensure that users were always prompted to the login page, I had modified the `mrbs_auth.inc` file to set `$result = 1` even for the `index.php` page, but now I also have the open, and this rule is no longer right. So, when it reached the line `if ($auth['deny_public_access']) { $result = max($result, 1); }`, it would always return 1. I just set the `result` for the index page back to `0` and now it works.

Sorry for the trouble, and thank you for your help.

Best regards
